### PR TITLE
Add functions to determine a gender given a name

### PR DESCRIPTION
**Pull request by Wallacoloo**
examples: is_gender("john", "male") == True. get_gender("Lucy") == "female". get_gender("Spiderman", default="superhero") == "superhero"

### CODE REVIEW

File: names/__init__.py [hunk 1]
```diff
@@ -41,3 +41,20 @@ def get_last_name():
...
+    with open(FILES['first:%s' % gender]) as name_file:
+        for line in name_file:
+            name, rank, cummulative, _ = line.split()
+            if name == first: return float(rank)
+    return default
+    
+def gender_ranks(name, default=0.):
+    return gender_rank(name, "male", default), gender_rank(name, "female", default)
+    
+def guess_gender(name, male="male", female="female", default=""):
```
Comment (treyhunner) [inline names/__init__.py:56]: The uses for all of the keyword arguments is uncertain to me.  Maybe this is a way to add i18n and support multiple languages?  In that case a translation file would probably be better.

Comment (treyhunner) [inline names/__init__.py:56]: I found that I needed to read each of these functions to determine their use.  For example I didn't realize that this function would return the gender that the name was identified with most commonly (which is relevant for a neutral name like "Taylor").  docstrings would help fix that problem.

File: names/__init__.py [hunk 2]
```diff
@@ -41,3 +41,20 @@ def get_last_name():
...
+    m, f = gender_ranks(name)
+    if m > f: return male
+    elif f > m: return female
+    else: return default
```
Comment (treyhunner) [inline names/__init__.py:60]: I'm not sure why this default string is better than returning None.

**Comment by treyhunner** [comment]
Thanks for the pull request.

I don't think I want to add this feature to my own branch of `names` because I don't think I will ever have a need for it and I don't understand the need in general.  Feel free to try to convince me if you'd like.

I made a few code critiques for you regardless.

**Comment by Wallacoloo** [comment]
Thanks for reviewing my code. The guess_gender function was just something I needed for my application. And the keyword arguments are just for convenience, much how python's dictionary object supports dict.get(key, default=None). They allow me to do things like: print "John skipped down the sidewalk on %s way to school." %guess_gender("John", male="his", female="her", default="the")

As far as the documentation goes... I was kinda following the standard you set xP

Anyways, if you don't want this functionality then I agree with your decision to keep these branches separate.

**Pull request closed without merging**